src/bgw/scheduler.c

```diff
@@ -571,9 +572,20 @@ start_scheduled_jobs(register_background_worker_callback_type bgw_register)
 	{
 		ScheduledBgwJob *sjob = lfirst(lc);
 
-		if (sjob->state == JOB_STATE_SCHEDULED &&
-			sjob->next_start <= ts_timer_get_current_timestamp())
+		int64 job_start_diff = sjob->next_start - ts_timer_get_current_timestamp();
+
+		if (sjob->state == JOB_STATE_SCHEDULED && job_start_diff < 0)
+		{
+			elog(DEBUG2, "starting scheduled job %d", sjob->job.fd.id);
 			scheduled_ts_bgw_job_start(sjob, bgw_register);
+		}
+		else
+		{
+			elog(DEBUG5,
```

Comment:
You might want to put these two messages at the same level. No strong opinion on how we should use the different debug levels.

Reply:
@mkindahl I used the different log levels intentionally. DEBUG2 for messages that only happen _sometimes_ and when a concrete action happens (e.g., the job is executed) and DEBUG5 for messages that are executed on each scheduler run.

Reply:
I have the same feeling. Why don't use the same debug level for all messages?

Reply:
The message with `DEBUG5` is produced for each job on each scheduler run. The message with `DEBUG2` is only produced every few minutes when the job is actually started. So, setting the log level to DEBUG2 should be enough to debug most problems and it would not flood the log file...

Reply:
This is fine for now, but we need to do some more work here to have guidelines for how to use this and also make sure to add debug printouts to more places (I just worked on a bug where it was impossible to follow the flow of a `compress_chunk` operation to see what went wrong).